internal/pkg/exec/docker.go

```diff
@@ -159,3 +161,38 @@ func imageName(uri, tag string) string {
 	}
 	return fmt.Sprintf("%s:%s", uri, tag)
 }
+
+// GetHelperProviderFromDockerCfg will fetch credential store used by docker from config file
+func (c DockerCommand) GetHelperProviderFromDockerCfg() (string) {
+	// Look into the default locations
+	pathsToTry := []string{"/.docker/config.json", ".dockercfg"}
+	for _, path := range pathsToTry {
+		content, err := ioutil.ReadFile(fmt.Sprintf("%s/%s",os.Getenv("HOME"), path))
```

Comment:
What do you think of storing the environment variable in the `DockerCommand` type to allow writing unit tests for this new method:
```go
// DockerCommand represents docker commands that can be run.
type DockerCommand struct {
	runner
	// Override in unit tests.
	homePath string // new field that can be substituted in tests
	buf *bytes.Buffer
}

// NewDockerCommand returns a DockerCommand.
func NewDockerCommand() DockerCommand {
	return DockerCommand{
		runner: NewCmd(),
		homePath: os.Getenv("HOME"),
	}
}
```

(for an example of unit tests with fake file systems, we use `afero.NewMemMapFs()`: https://github.com/aws/copilot-cli/blob/c2d143dfe781a865ed823984c88065a92836b4c6/internal/pkg/workspace/workspace_test.go#L123-L126)

Reply:
Makes sense and thanks for the suggestion. I have added additional test cases to unit test this method